// @lukso/lsp1delegate-contracts/dist/abi.mjs
const lsp1UniversalReceiverDelegateUpAbi = [
  {
    type: "error",
    inputs: [{ name: "caller", internalType: "address", type: "address" }],
    name: "CannotRegisterEOAsAsAssets"
  },
  {
    type: "function",
    inputs: [],
    name: "VERSION",
    outputs: [{ name: "", internalType: "string", type: "string" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "notifier", internalType: "address", type: "address" },
      { name: "", internalType: "uint256", type: "uint256" },
      { name: "typeId", internalType: "bytes32", type: "bytes32" },
      { name: "", internalType: "bytes", type: "bytes" }
    ],
    name: "universalReceiverDelegate",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "nonpayable"
  }
];
const lsp1UniversalReceiverDelegateVaultAbi = [
  {
    type: "error",
    inputs: [{ name: "caller", internalType: "address", type: "address" }],
    name: "CannotRegisterEOAsAsAssets"
  },
  {
    type: "function",
    inputs: [],
    name: "VERSION",
    outputs: [{ name: "", internalType: "string", type: "string" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "notifier", internalType: "address", type: "address" },
      { name: "", internalType: "uint256", type: "uint256" },
      { name: "typeId", internalType: "bytes32", type: "bytes32" },
      { name: "", internalType: "bytes", type: "bytes" }
    ],
    name: "universalReceiverDelegate",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "nonpayable"
  }
];

export { lsp1UniversalReceiverDelegateUpAbi, lsp1UniversalReceiverDelegateVaultAbi };
