// @lukso/lsp9-contracts/dist/abi.cjs
'use strict';

const lsp9VaultAbi = [
  {
    type: "constructor",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    stateMutability: "payable"
  },
  { type: "error", inputs: [], name: "ERC725X_ContractDeploymentFailed" },
  {
    type: "error",
    inputs: [],
    name: "ERC725X_CreateOperationsRequireEmptyRecipientAddress"
  },
  { type: "error", inputs: [], name: "ERC725X_ExecuteParametersEmptyArray" },
  {
    type: "error",
    inputs: [],
    name: "ERC725X_ExecuteParametersLengthMismatch"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "value", internalType: "uint256", type: "uint256" }
    ],
    name: "ERC725X_InsufficientBalance"
  },
  { type: "error", inputs: [], name: "ERC725X_MsgValueDisallowedInStaticCall" },
  { type: "error", inputs: [], name: "ERC725X_NoContractBytecodeProvided" },
  {
    type: "error",
    inputs: [
      {
        name: "operationTypeProvided",
        internalType: "uint256",
        type: "uint256"
      }
    ],
    name: "ERC725X_UnknownOperationType"
  },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "caller", internalType: "address", type: "address" }],
    name: "LSP14CallerNotPendingOwner"
  },
  { type: "error", inputs: [], name: "LSP14CannotTransferOwnershipToSelf" },
  {
    type: "error",
    inputs: [],
    name: "LSP14MustAcceptOwnershipInSeparateTransaction"
  },
  {
    type: "error",
    inputs: [
      {
        name: "renounceOwnershipStart",
        internalType: "uint256",
        type: "uint256"
      },
      {
        name: "renounceOwnershipEnd",
        internalType: "uint256",
        type: "uint256"
      }
    ],
    name: "LSP14NotInRenounceOwnershipInterval"
  },
  {
    type: "error",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "LSP1DelegateNotAllowedToSetDataKey"
  },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [
      { name: "callerAddress", internalType: "address", type: "address" }
    ],
    name: "OwnableCallerNotTheOwner"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operationType",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "contractAddress",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "salt", internalType: "bytes32", type: "bytes32", indexed: true }
    ],
    name: "ContractCreated"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operationType",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "target",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      {
        name: "selector",
        internalType: "bytes4",
        type: "bytes4",
        indexed: true
      }
    ],
    name: "Executed"
  },
  { type: "event", anonymous: false, inputs: [], name: "OwnershipRenounced" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferStarted"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [],
    name: "RenounceOwnershipStarted"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "from", internalType: "address", type: "address", indexed: true },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "typeId",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "receivedData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      },
      {
        name: "returnedValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "UniversalReceiver"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [],
    name: "RENOUNCE_OWNERSHIP_CONFIRMATION_DELAY",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "RENOUNCE_OWNERSHIP_CONFIRMATION_PERIOD",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "VERSION",
    outputs: [{ name: "", internalType: "string", type: "string" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "acceptOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operationType", internalType: "uint256", type: "uint256" },
      { name: "target", internalType: "address", type: "address" },
      { name: "value", internalType: "uint256", type: "uint256" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "execute",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "operationsType", internalType: "uint256[]", type: "uint256[]" },
      { name: "targets", internalType: "address[]", type: "address[]" },
      { name: "values", internalType: "uint256[]", type: "uint256[]" },
      { name: "datas", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "executeBatch",
    outputs: [{ name: "", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "pendingOwner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "typeId", internalType: "bytes32", type: "bytes32" },
      { name: "receivedData", internalType: "bytes", type: "bytes" }
    ],
    name: "universalReceiver",
    outputs: [{ name: "returnedValues", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp9VaultInitAbi = [
  { type: "constructor", inputs: [], stateMutability: "nonpayable" },
  { type: "error", inputs: [], name: "ERC725X_ContractDeploymentFailed" },
  {
    type: "error",
    inputs: [],
    name: "ERC725X_CreateOperationsRequireEmptyRecipientAddress"
  },
  { type: "error", inputs: [], name: "ERC725X_ExecuteParametersEmptyArray" },
  {
    type: "error",
    inputs: [],
    name: "ERC725X_ExecuteParametersLengthMismatch"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "value", internalType: "uint256", type: "uint256" }
    ],
    name: "ERC725X_InsufficientBalance"
  },
  { type: "error", inputs: [], name: "ERC725X_MsgValueDisallowedInStaticCall" },
  { type: "error", inputs: [], name: "ERC725X_NoContractBytecodeProvided" },
  {
    type: "error",
    inputs: [
      {
        name: "operationTypeProvided",
        internalType: "uint256",
        type: "uint256"
      }
    ],
    name: "ERC725X_UnknownOperationType"
  },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "caller", internalType: "address", type: "address" }],
    name: "LSP14CallerNotPendingOwner"
  },
  { type: "error", inputs: [], name: "LSP14CannotTransferOwnershipToSelf" },
  {
    type: "error",
    inputs: [],
    name: "LSP14MustAcceptOwnershipInSeparateTransaction"
  },
  {
    type: "error",
    inputs: [
      {
        name: "renounceOwnershipStart",
        internalType: "uint256",
        type: "uint256"
      },
      {
        name: "renounceOwnershipEnd",
        internalType: "uint256",
        type: "uint256"
      }
    ],
    name: "LSP14NotInRenounceOwnershipInterval"
  },
  {
    type: "error",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "LSP1DelegateNotAllowedToSetDataKey"
  },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [
      { name: "callerAddress", internalType: "address", type: "address" }
    ],
    name: "OwnableCallerNotTheOwner"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operationType",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "contractAddress",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "salt", internalType: "bytes32", type: "bytes32", indexed: true }
    ],
    name: "ContractCreated"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operationType",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "target",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      {
        name: "selector",
        internalType: "bytes4",
        type: "bytes4",
        indexed: true
      }
    ],
    name: "Executed"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "version", internalType: "uint8", type: "uint8", indexed: false }
    ],
    name: "Initialized"
  },
  { type: "event", anonymous: false, inputs: [], name: "OwnershipRenounced" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferStarted"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [],
    name: "RenounceOwnershipStarted"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "from", internalType: "address", type: "address", indexed: true },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "typeId",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "receivedData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      },
      {
        name: "returnedValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "UniversalReceiver"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [],
    name: "RENOUNCE_OWNERSHIP_CONFIRMATION_DELAY",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "RENOUNCE_OWNERSHIP_CONFIRMATION_PERIOD",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "VERSION",
    outputs: [{ name: "", internalType: "string", type: "string" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "acceptOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operationType", internalType: "uint256", type: "uint256" },
      { name: "target", internalType: "address", type: "address" },
      { name: "value", internalType: "uint256", type: "uint256" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "execute",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "operationsType", internalType: "uint256[]", type: "uint256[]" },
      { name: "targets", internalType: "address[]", type: "address[]" },
      { name: "values", internalType: "uint256[]", type: "uint256[]" },
      { name: "datas", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "executeBatch",
    outputs: [{ name: "", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "initialize",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "pendingOwner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "typeId", internalType: "bytes32", type: "bytes32" },
      { name: "receivedData", internalType: "bytes", type: "bytes" }
    ],
    name: "universalReceiver",
    outputs: [{ name: "returnedValues", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  { type: "receive", stateMutability: "payable" }
];

exports.lsp9VaultAbi = lsp9VaultAbi;
exports.lsp9VaultInitAbi = lsp9VaultInitAbi;
