// @lukso/lsp20-contracts/dist/abi.mjs
const ilsp20CallVerifierAbi = [
  {
    type: "function",
    inputs: [
      { name: "requestor", internalType: "address", type: "address" },
      { name: "target", internalType: "address", type: "address" },
      { name: "caller", internalType: "address", type: "address" },
      { name: "value", internalType: "uint256", type: "uint256" },
      { name: "callData", internalType: "bytes", type: "bytes" }
    ],
    name: "lsp20VerifyCall",
    outputs: [
      { name: "returnedStatus", internalType: "bytes4", type: "bytes4" }
    ],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "callHash", internalType: "bytes32", type: "bytes32" },
      { name: "callResult", internalType: "bytes", type: "bytes" }
    ],
    name: "lsp20VerifyCallResult",
    outputs: [{ name: "", internalType: "bytes4", type: "bytes4" }],
    stateMutability: "nonpayable"
  }
];
const lsp20CallVerificationAbi = [];

export { ilsp20CallVerifierAbi, lsp20CallVerificationAbi };
