// @lukso/lsp25-contracts/dist/abi.mjs
const ilsp25ExecuteRelayCallAbi = [
  {
    type: "function",
    inputs: [
      { name: "signature", internalType: "bytes", type: "bytes" },
      { name: "nonce", internalType: "uint256", type: "uint256" },
      { name: "validityTimestamps", internalType: "uint256", type: "uint256" },
      { name: "payload", internalType: "bytes", type: "bytes" }
    ],
    name: "executeRelayCall",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "signatures", internalType: "bytes[]", type: "bytes[]" },
      { name: "nonces", internalType: "uint256[]", type: "uint256[]" },
      {
        name: "validityTimestamps",
        internalType: "uint256[]",
        type: "uint256[]"
      },
      { name: "values", internalType: "uint256[]", type: "uint256[]" },
      { name: "payloads", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "executeRelayCallBatch",
    outputs: [{ name: "", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "channelId", internalType: "uint128", type: "uint128" }
    ],
    name: "getNonce",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  }
];

export { ilsp25ExecuteRelayCallAbi };
