// @lukso/lsp4-contracts/dist/index.mjs
const LSP4SupportedStandard = {
  key: "0xeafec4d89fa9619884b60000a4d96624a38f7ac2d8d9a604ecf07c12c77e480c",
  value: "0xa4d96624"
};
const LSP4DataKeys = {
  SupportedStandards_LSP4: LSP4SupportedStandard.key,
  // keccak256('LSP4TokenName')
  LSP4TokenName: "0xdeba1e292f8ba88238e10ab3c7f88bd4be4fac56cad5194b6ecceaf653468af1",
  // keccak256('LSP4TokenSymbol')
  LSP4TokenSymbol: "0x2f0a68ab07768e01943a599e73362a0e17a63a72e94dd2e384d2c1d4db932756",
  // keccak256('LSP4TokenType)
  LSP4TokenType: "0xe0261fa95db2eb3b5439bd033cda66d56b96f92f243a8228fd87550ed7bdfdb3",
  // keccak256('LSP4Metadata')
  LSP4Metadata: "0x9afb95cacc9f95858ec44aa8c3b685511002e30ae54415823f406128b85b238e",
  // LSP4CreatorsMap:<address>  + bytes2(0)
  LSP4CreatorsMap: "0x6de85eaf5d982b4e5da00000",
  // keccak256('"LSP4Creators[]')
  "LSP4Creators[]": {
    length: "0x114bd03b3a46d48759680d81ebb2b414fda7d030a7105a851867accf1c2352e7",
    index: "0x114bd03b3a46d48759680d81ebb2b414"
  }
};
const LSP4_TOKEN_TYPES = {
  TOKEN: 0,
  NFT: 1,
  COLLECTION: 2
};

export { LSP4DataKeys, LSP4SupportedStandard, LSP4_TOKEN_TYPES };
