// @lukso/lsp16-contracts/dist/abi.mjs
const lsp16UniversalFactoryAbi = [
  { type: "error", inputs: [], name: "ContractInitializationFailed" },
  { type: "error", inputs: [], name: "InvalidValueSum" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "createdContract",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "providedSalt",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "generatedSalt",
        internalType: "bytes32",
        type: "bytes32",
        indexed: false
      },
      {
        name: "initialized",
        internalType: "bool",
        type: "bool",
        indexed: true
      },
      {
        name: "initializeCalldata",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "ContractCreated"
  },
  {
    type: "function",
    inputs: [
      {
        name: "creationBytecodeHash",
        internalType: "bytes32",
        type: "bytes32"
      },
      { name: "providedSalt", internalType: "bytes32", type: "bytes32" },
      { name: "initializable", internalType: "bool", type: "bool" },
      { name: "initializeCalldata", internalType: "bytes", type: "bytes" }
    ],
    name: "computeAddress",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      {
        name: "implementationContract",
        internalType: "address",
        type: "address"
      },
      { name: "providedSalt", internalType: "bytes32", type: "bytes32" },
      { name: "initializable", internalType: "bool", type: "bool" },
      { name: "initializeCalldata", internalType: "bytes", type: "bytes" }
    ],
    name: "computeERC1167Address",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "creationBytecode", internalType: "bytes", type: "bytes" },
      { name: "providedSalt", internalType: "bytes32", type: "bytes32" }
    ],
    name: "deployCreate2",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "creationBytecode", internalType: "bytes", type: "bytes" },
      { name: "providedSalt", internalType: "bytes32", type: "bytes32" },
      { name: "initializeCalldata", internalType: "bytes", type: "bytes" },
      { name: "constructorMsgValue", internalType: "uint256", type: "uint256" },
      {
        name: "initializeCalldataMsgValue",
        internalType: "uint256",
        type: "uint256"
      }
    ],
    name: "deployCreate2AndInitialize",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      {
        name: "implementationContract",
        internalType: "address",
        type: "address"
      },
      { name: "providedSalt", internalType: "bytes32", type: "bytes32" }
    ],
    name: "deployERC1167Proxy",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      {
        name: "implementationContract",
        internalType: "address",
        type: "address"
      },
      { name: "providedSalt", internalType: "bytes32", type: "bytes32" },
      { name: "initializeCalldata", internalType: "bytes", type: "bytes" }
    ],
    name: "deployERC1167ProxyAndInitialize",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "providedSalt", internalType: "bytes32", type: "bytes32" },
      { name: "initializable", internalType: "bool", type: "bool" },
      { name: "initializeCalldata", internalType: "bytes", type: "bytes" }
    ],
    name: "generateSalt",
    outputs: [{ name: "", internalType: "bytes32", type: "bytes32" }],
    stateMutability: "pure"
  }
];

export { lsp16UniversalFactoryAbi };
