// @lukso/lsp6-contracts/dist/abi.cjs
'use strict';

const ilsp6KeyManagerAbi = [
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "signer",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "selector",
        internalType: "bytes4",
        type: "bytes4",
        indexed: true
      }
    ],
    name: "PermissionsVerified"
  },
  {
    type: "function",
    inputs: [{ name: "payload", internalType: "bytes", type: "bytes" }],
    name: "execute",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "values", internalType: "uint256[]", type: "uint256[]" },
      { name: "payloads", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "executeBatch",
    outputs: [{ name: "", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "hash", internalType: "bytes32", type: "bytes32" },
      { name: "signature", internalType: "bytes", type: "bytes" }
    ],
    name: "isValidSignature",
    outputs: [{ name: "magicValue", internalType: "bytes4", type: "bytes4" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "target",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  }
];
const lsp6KeyManagerAbi = [
  {
    type: "constructor",
    inputs: [{ name: "target_", internalType: "address", type: "address" }],
    stateMutability: "nonpayable"
  },
  { type: "error", inputs: [], name: "BatchExecuteParamsLengthMismatch" },
  {
    type: "error",
    inputs: [],
    name: "BatchExecuteRelayCallParamsLengthMismatch"
  },
  { type: "error", inputs: [], name: "CallingKeyManagerNotAllowed" },
  { type: "error", inputs: [], name: "DelegateCallDisallowedViaKeyManager" },
  { type: "error", inputs: [], name: "ERC725X_ExecuteParametersEmptyArray" },
  {
    type: "error",
    inputs: [],
    name: "ERC725X_ExecuteParametersLengthMismatch"
  },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  {
    type: "error",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "InvalidDataValuesForDataKeys"
  },
  {
    type: "error",
    inputs: [
      { name: "invalidFunction", internalType: "bytes4", type: "bytes4" }
    ],
    name: "InvalidERC725Function"
  },
  {
    type: "error",
    inputs: [
      { name: "allowedCallsValue", internalType: "bytes", type: "bytes" }
    ],
    name: "InvalidEncodedAllowedCalls"
  },
  {
    type: "error",
    inputs: [
      { name: "value", internalType: "bytes", type: "bytes" },
      { name: "context", internalType: "string", type: "string" }
    ],
    name: "InvalidEncodedAllowedERC725YDataKeys"
  },
  { type: "error", inputs: [], name: "InvalidLSP6Target" },
  {
    type: "error",
    inputs: [{ name: "payload", internalType: "bytes", type: "bytes" }],
    name: "InvalidPayload"
  },
  {
    type: "error",
    inputs: [
      { name: "signer", internalType: "address", type: "address" },
      { name: "invalidNonce", internalType: "uint256", type: "uint256" },
      { name: "signature", internalType: "bytes", type: "bytes" }
    ],
    name: "InvalidRelayNonce"
  },
  {
    type: "error",
    inputs: [{ name: "from", internalType: "address", type: "address" }],
    name: "InvalidWhitelistedCall"
  },
  {
    type: "error",
    inputs: [],
    name: "KeyManagerCannotBeSetAsExtensionForLSP20Functions"
  },
  {
    type: "error",
    inputs: [
      { name: "totalValues", internalType: "uint256", type: "uint256" },
      { name: "msgValue", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP6BatchExcessiveValueSent"
  },
  {
    type: "error",
    inputs: [
      { name: "totalValues", internalType: "uint256", type: "uint256" },
      { name: "msgValue", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP6BatchInsufficientValueSent"
  },
  {
    type: "error",
    inputs: [{ name: "from", internalType: "address", type: "address" }],
    name: "NoCallsAllowed"
  },
  {
    type: "error",
    inputs: [{ name: "from", internalType: "address", type: "address" }],
    name: "NoERC725YDataKeysAllowed"
  },
  {
    type: "error",
    inputs: [{ name: "from", internalType: "address", type: "address" }],
    name: "NoPermissionsSet"
  },
  {
    type: "error",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "selector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NotAllowedCall"
  },
  {
    type: "error",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "disallowedKey", internalType: "bytes32", type: "bytes32" }
    ],
    name: "NotAllowedERC725YDataKey"
  },
  {
    type: "error",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "permission", internalType: "string", type: "string" }
    ],
    name: "NotAuthorised"
  },
  {
    type: "error",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "NotRecognisedPermissionKey"
  },
  { type: "error", inputs: [], name: "RelayCallBeforeStartTime" },
  { type: "error", inputs: [], name: "RelayCallExpired" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "signer",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "selector",
        internalType: "bytes4",
        type: "bytes4",
        indexed: true
      }
    ],
    name: "PermissionsVerified"
  },
  {
    type: "function",
    inputs: [],
    name: "VERSION",
    outputs: [{ name: "", internalType: "string", type: "string" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "payload", internalType: "bytes", type: "bytes" }],
    name: "execute",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "values", internalType: "uint256[]", type: "uint256[]" },
      { name: "payloads", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "executeBatch",
    outputs: [{ name: "", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "signature", internalType: "bytes", type: "bytes" },
      { name: "nonce", internalType: "uint256", type: "uint256" },
      { name: "validityTimestamps", internalType: "uint256", type: "uint256" },
      { name: "payload", internalType: "bytes", type: "bytes" }
    ],
    name: "executeRelayCall",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "signatures", internalType: "bytes[]", type: "bytes[]" },
      { name: "nonces", internalType: "uint256[]", type: "uint256[]" },
      {
        name: "validityTimestamps",
        internalType: "uint256[]",
        type: "uint256[]"
      },
      { name: "values", internalType: "uint256[]", type: "uint256[]" },
      { name: "payloads", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "executeRelayCallBatch",
    outputs: [{ name: "", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "channelId", internalType: "uint128", type: "uint128" }
    ],
    name: "getNonce",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataHash", internalType: "bytes32", type: "bytes32" },
      { name: "signature", internalType: "bytes", type: "bytes" }
    ],
    name: "isValidSignature",
    outputs: [
      { name: "returnedStatus", internalType: "bytes4", type: "bytes4" }
    ],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "", internalType: "address", type: "address" },
      { name: "targetContract", internalType: "address", type: "address" },
      { name: "caller", internalType: "address", type: "address" },
      { name: "msgValue", internalType: "uint256", type: "uint256" },
      { name: "callData", internalType: "bytes", type: "bytes" }
    ],
    name: "lsp20VerifyCall",
    outputs: [{ name: "", internalType: "bytes4", type: "bytes4" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "", internalType: "bytes32", type: "bytes32" },
      { name: "", internalType: "bytes", type: "bytes" }
    ],
    name: "lsp20VerifyCallResult",
    outputs: [{ name: "", internalType: "bytes4", type: "bytes4" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "target",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  }
];
const lsp6KeyManagerInitAbi = [
  { type: "constructor", inputs: [], stateMutability: "nonpayable" },
  { type: "error", inputs: [], name: "BatchExecuteParamsLengthMismatch" },
  {
    type: "error",
    inputs: [],
    name: "BatchExecuteRelayCallParamsLengthMismatch"
  },
  { type: "error", inputs: [], name: "CallingKeyManagerNotAllowed" },
  { type: "error", inputs: [], name: "DelegateCallDisallowedViaKeyManager" },
  { type: "error", inputs: [], name: "ERC725X_ExecuteParametersEmptyArray" },
  {
    type: "error",
    inputs: [],
    name: "ERC725X_ExecuteParametersLengthMismatch"
  },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  {
    type: "error",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "InvalidDataValuesForDataKeys"
  },
  {
    type: "error",
    inputs: [
      { name: "invalidFunction", internalType: "bytes4", type: "bytes4" }
    ],
    name: "InvalidERC725Function"
  },
  {
    type: "error",
    inputs: [
      { name: "allowedCallsValue", internalType: "bytes", type: "bytes" }
    ],
    name: "InvalidEncodedAllowedCalls"
  },
  {
    type: "error",
    inputs: [
      { name: "value", internalType: "bytes", type: "bytes" },
      { name: "context", internalType: "string", type: "string" }
    ],
    name: "InvalidEncodedAllowedERC725YDataKeys"
  },
  { type: "error", inputs: [], name: "InvalidLSP6Target" },
  {
    type: "error",
    inputs: [{ name: "payload", internalType: "bytes", type: "bytes" }],
    name: "InvalidPayload"
  },
  {
    type: "error",
    inputs: [
      { name: "signer", internalType: "address", type: "address" },
      { name: "invalidNonce", internalType: "uint256", type: "uint256" },
      { name: "signature", internalType: "bytes", type: "bytes" }
    ],
    name: "InvalidRelayNonce"
  },
  {
    type: "error",
    inputs: [{ name: "from", internalType: "address", type: "address" }],
    name: "InvalidWhitelistedCall"
  },
  {
    type: "error",
    inputs: [],
    name: "KeyManagerCannotBeSetAsExtensionForLSP20Functions"
  },
  {
    type: "error",
    inputs: [
      { name: "totalValues", internalType: "uint256", type: "uint256" },
      { name: "msgValue", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP6BatchExcessiveValueSent"
  },
  {
    type: "error",
    inputs: [
      { name: "totalValues", internalType: "uint256", type: "uint256" },
      { name: "msgValue", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP6BatchInsufficientValueSent"
  },
  {
    type: "error",
    inputs: [{ name: "from", internalType: "address", type: "address" }],
    name: "NoCallsAllowed"
  },
  {
    type: "error",
    inputs: [{ name: "from", internalType: "address", type: "address" }],
    name: "NoERC725YDataKeysAllowed"
  },
  {
    type: "error",
    inputs: [{ name: "from", internalType: "address", type: "address" }],
    name: "NoPermissionsSet"
  },
  {
    type: "error",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "selector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NotAllowedCall"
  },
  {
    type: "error",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "disallowedKey", internalType: "bytes32", type: "bytes32" }
    ],
    name: "NotAllowedERC725YDataKey"
  },
  {
    type: "error",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "permission", internalType: "string", type: "string" }
    ],
    name: "NotAuthorised"
  },
  {
    type: "error",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "NotRecognisedPermissionKey"
  },
  { type: "error", inputs: [], name: "RelayCallBeforeStartTime" },
  { type: "error", inputs: [], name: "RelayCallExpired" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "version", internalType: "uint8", type: "uint8", indexed: false }
    ],
    name: "Initialized"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "signer",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "selector",
        internalType: "bytes4",
        type: "bytes4",
        indexed: true
      }
    ],
    name: "PermissionsVerified"
  },
  {
    type: "function",
    inputs: [],
    name: "VERSION",
    outputs: [{ name: "", internalType: "string", type: "string" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "payload", internalType: "bytes", type: "bytes" }],
    name: "execute",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "values", internalType: "uint256[]", type: "uint256[]" },
      { name: "payloads", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "executeBatch",
    outputs: [{ name: "", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "signature", internalType: "bytes", type: "bytes" },
      { name: "nonce", internalType: "uint256", type: "uint256" },
      { name: "validityTimestamps", internalType: "uint256", type: "uint256" },
      { name: "payload", internalType: "bytes", type: "bytes" }
    ],
    name: "executeRelayCall",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "signatures", internalType: "bytes[]", type: "bytes[]" },
      { name: "nonces", internalType: "uint256[]", type: "uint256[]" },
      {
        name: "validityTimestamps",
        internalType: "uint256[]",
        type: "uint256[]"
      },
      { name: "values", internalType: "uint256[]", type: "uint256[]" },
      { name: "payloads", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "executeRelayCallBatch",
    outputs: [{ name: "", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "channelId", internalType: "uint128", type: "uint128" }
    ],
    name: "getNonce",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "target_", internalType: "address", type: "address" }],
    name: "initialize",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataHash", internalType: "bytes32", type: "bytes32" },
      { name: "signature", internalType: "bytes", type: "bytes" }
    ],
    name: "isValidSignature",
    outputs: [
      { name: "returnedStatus", internalType: "bytes4", type: "bytes4" }
    ],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "", internalType: "address", type: "address" },
      { name: "targetContract", internalType: "address", type: "address" },
      { name: "caller", internalType: "address", type: "address" },
      { name: "msgValue", internalType: "uint256", type: "uint256" },
      { name: "callData", internalType: "bytes", type: "bytes" }
    ],
    name: "lsp20VerifyCall",
    outputs: [{ name: "", internalType: "bytes4", type: "bytes4" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "", internalType: "bytes32", type: "bytes32" },
      { name: "", internalType: "bytes", type: "bytes" }
    ],
    name: "lsp20VerifyCallResult",
    outputs: [{ name: "", internalType: "bytes4", type: "bytes4" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "target",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  }
];

exports.ilsp6KeyManagerAbi = ilsp6KeyManagerAbi;
exports.lsp6KeyManagerAbi = lsp6KeyManagerAbi;
exports.lsp6KeyManagerInitAbi = lsp6KeyManagerInitAbi;
