// @lukso/lsp1-contracts/dist/abi.cjs
'use strict';

const ilsp1UniversalReceiverAbi = [
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "from", internalType: "address", type: "address", indexed: true },
      {
        name: "value",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "typeId",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "receivedData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      },
      {
        name: "returnedValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "UniversalReceiver"
  },
  {
    type: "function",
    inputs: [
      { name: "typeId", internalType: "bytes32", type: "bytes32" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "universalReceiver",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  }
];
const ilsp1UniversalReceiverDelegateAbi = [
  {
    type: "function",
    inputs: [
      { name: "sender", internalType: "address", type: "address" },
      { name: "value", internalType: "uint256", type: "uint256" },
      { name: "typeId", internalType: "bytes32", type: "bytes32" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "universalReceiverDelegate",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "nonpayable"
  }
];

exports.ilsp1UniversalReceiverAbi = ilsp1UniversalReceiverAbi;
exports.ilsp1UniversalReceiverDelegateAbi = ilsp1UniversalReceiverDelegateAbi;
