// @lukso/lsp14-contracts/dist/abi.cjs
'use strict';

const ilsp14Ownable2StepAbi = [
  { type: "event", anonymous: false, inputs: [], name: "OwnershipRenounced" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferStarted"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [],
    name: "RenounceOwnershipStarted"
  },
  {
    type: "function",
    inputs: [],
    name: "acceptOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "pendingOwner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  }
];
const lsp14Ownable2StepAbi = [
  {
    type: "error",
    inputs: [{ name: "caller", internalType: "address", type: "address" }],
    name: "LSP14CallerNotPendingOwner"
  },
  { type: "error", inputs: [], name: "LSP14CannotTransferOwnershipToSelf" },
  {
    type: "error",
    inputs: [],
    name: "LSP14MustAcceptOwnershipInSeparateTransaction"
  },
  {
    type: "error",
    inputs: [
      {
        name: "renounceOwnershipStart",
        internalType: "uint256",
        type: "uint256"
      },
      {
        name: "renounceOwnershipEnd",
        internalType: "uint256",
        type: "uint256"
      }
    ],
    name: "LSP14NotInRenounceOwnershipInterval"
  },
  {
    type: "error",
    inputs: [
      { name: "callerAddress", internalType: "address", type: "address" }
    ],
    name: "OwnableCallerNotTheOwner"
  },
  { type: "event", anonymous: false, inputs: [], name: "OwnershipRenounced" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferStarted"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [],
    name: "RenounceOwnershipStarted"
  },
  {
    type: "function",
    inputs: [],
    name: "RENOUNCE_OWNERSHIP_CONFIRMATION_DELAY",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "RENOUNCE_OWNERSHIP_CONFIRMATION_PERIOD",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "acceptOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "pendingOwner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  }
];

exports.ilsp14Ownable2StepAbi = ilsp14Ownable2StepAbi;
exports.lsp14Ownable2StepAbi = lsp14Ownable2StepAbi;
